src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -204,6 +204,11 @@ static constexpr auto BLOOM_FILTER_MASK = BLOOM_FILTER_SIZE - 1;
 // constructor does not do what you would expect it to, so try to avoid it.
 using BloomFilter = boost::dynamic_bitset<>;
 
+// EMPTY_BLOOM_FILTER is initialized by creating a BloomFilter with every value being false and using bitwise
+// negation (~x). As the negation is surprisingly expensive, we create a static empty bloom filter and reference
+// it where needed.
+static const auto EMPTY_BLOOM_FILTER = ~BloomFilter(BLOOM_FILTER_SIZE);
```

Comment:
Why do we need this reference with a default value and do not use an std::optional?

Reply:
Because it would introduce another branch in the scaldingly hot materialization loop

Reply:
I think a short comment about that help me a lot.

Reply:
In general it's not easy to follow. An empty bloom means that all materialization is skipped. When can that be the case?

Reply:
Better to call it `ALL_TRUE_BLOOM_FILTER`?

Reply:
Ahhh, I misunderstood your explanation and just focussed on `EMPTY_...`. Yes, `ALL_TRUE` is much better.